docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+# What is Jupyter and JupyterHub?
+
+JupyterHub is not what you think it is.  Most things you think are
+part of JupyterHub are actually handled by some other component, for
+example the spawner or notebook server itself, and it's not always
+obvious how the parts relate.  The knowledge contained here hasn't
+been assembled in one place before, and is essential to understand
+when setting up a sufficiently complex Jupyter(Hub) setup.
+
+This document was originally written to assist in debugging: very
+often, the actual problem is not where one thinks it is and thus
+people can't easily debug.  In order to tell this story, we start at
+JupyterHub and go all the way down to the fundamental components of
+Jupyter.
+
+In this document, we occasionally leave things out or bend the truth
+where it helps in explanation, and give our explanations in terms of
+Python even though Jupyter itself is language-neutral.  The "(&)"
```

Comment:
Does the `&` symbol have a functional use on the page? (e.g., does it create a hyperlink or something like that?) if not, I think we should either:

1. Turn these into footnotes or in-line links to other sections
2. Remove the `&` symbol because I think many readers won't really understand what to do with it

Reply:
I got this idea form `regex(7)`, which uses `(!)` to indicate a certain thing that is commonly said but shouldn't be spelled out each time (possibly non-portable decisions).  I thought I had two choices: a) say thing that are not entirely true or the whole story or b) very often say "but there is more/this is a simplification".  Perhaps the definition of `(&)` could be c) emphasized in an admonition, or d) it could be changed to "everything on this page may be simplified and expect some inaccuracies here".  But I'd expect (d) could lead to lots of updates adding more details, making the page too long for it's main purpose...